targets/csharp/source/source/PlayFabGameServerModels.cs

```diff
@@ -0,0 +1,184 @@
+
+        public GameServerConfiguration()
+        {
+            GameCertificates = new Dictionary<string, string>();
+            BuildMetadata = new Dictionary<string, string>();
+            GamePorts = new Dictionary<string, string>();
+        }
+    }
+
+    class EnvironmentVariableConfiguration : GameServerConfiguration
+    {
+        public EnvironmentVariableConfiguration() : base()
+        {
+            HeartbeatEndpoint = Environment.GetEnvironmentVariable(HEARTBEAT_ENDPOINT_ENV_VAR);
+            ServerId = Environment.GetEnvironmentVariable(SERVER_ID_ENV_VAR);
+            LogFolder = Environment.GetEnvironmentVariable(LOG_FOLDER_ENV_VAR);
+            SharedContentFolder = Environment.GetEnvironmentVariable(SHARED_CONTENT_FOLDER_ENV_VAR);
+            if (string.IsNullOrWhiteSpace(HeartbeatEndpoint) || string.IsNullOrWhiteSpace(ServerId))
+            {
+                throw new GSDKInitializationException("Heartbeat endpoint and Server id are required configuration values.");
+            }
+        }
+    }
+
+    class JsonFileConfiguration : GameServerConfiguration
+    {
+        private bool _shouldLog;
+
+        public override bool ShouldLog()
+        {
+            return _shouldLog;
+        }
+
+        public JsonFileConfiguration(ISerializerPlugin jsonSerializer, string fileName) : base()
+        {
+            try
+            {
+                using (StreamReader reader = File.OpenText(fileName))
+                {
+                    JsonGsdkSchema config = jsonSerializer.DeserializeObject<JsonGsdkSchema>(reader.ReadToEnd());
+
+                    HeartbeatEndpoint = config.HeartbeatEndpoint;
+                    ServerId = config.SessionHostId;
+                    LogFolder = config.LogFolder;
+                    SharedContentFolder = config.SharedContentFolder;
+                    CertificateFolder = config.CertificateFolder;
+                    GameCertificates = config.GameCertificates ?? new Dictionary<string, string>();
+                    GamePorts = config.GamePorts ?? new Dictionary<string, string>();
+                    BuildMetadata = config.BuildMetadata ?? new Dictionary<string, string>();
+                    _shouldLog = config.ShouldLog.HasValue && config.ShouldLog.Value;
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new GSDKInitializationException($"Cannot read configuration file {fileName}", ex);
+            }
+            
+            if (string.IsNullOrWhiteSpace(HeartbeatEndpoint) || string.IsNullOrWhiteSpace(ServerId))
+            {
+                throw new GSDKInitializationException($"Heartbeat endpoint and Server id are required configuration values. Please double check configuration file at: {fileName}");
+            }
+        }
+    }
+
+    class JsonGsdkSchema
+    {
+        [JsonProperty(PropertyName = "heartbeatEndpoint")]
+        public string HeartbeatEndpoint { get; set; }
```

Comment:
Please don't use properties when it's not used for a purpose.
Fields are about 1000x more performant than properties.

Reply:
(applies to this whole file)

Reply:
I got rid of them for the JsonSchema, but for stuff that is customer facing I'll keep them around

---
In reply to: [256126324](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256126324) [](ancestors = 256126324)